Feature: Support TYPO3 8

DanielSiepmann opened the pull request:
Update tests and code base to Support TYPO3 CMS 8.

Resolves: #55

DanielSiepmann committed:
TASK: Update composer

Require TYPO3 CMS 8.2 and PHP 7.1 as minimum.
Use conventionally lower case web folder.

DanielSiepmann committed:
WIP|TASK: Use new TYPO3

Use testing framework configuration.
Use new db.
Begin with replacement of old TYPO3_DB.

DanielSiepmann committed:
Merge remote-tracking branch 'origin/develop' into feature/cms-8-support

DanielSiepmann committed:
TASK: Migrate dev dependencies

As testing framework is used, we can prefer dist again to speed up
composer installation.

DanielSiepmann committed:
BUGFIX: Allow tests to run without database connection

Ad TYPO3 Core now makes use of Doctrine, a connection is required to
build system where. Therefore we move it to an own method to exchange
the execution inside of tests.

DanielSiepmann committed:
TASK: Make extension more compatible

Migrate sql to doctrine.
Migrate relation resolver to use new API.

DanielSiepmann committed:
TASK: Finish TYPO3 CMS 8 update

DanielSiepmann committed:
TASK: Update dependencies for CMS 8

DanielSiepmann committed:
BUGFIX: Fix TypoScript warning

Constants can't use multi line strings.

DanielSiepmann committed:
BUGFIX: Fix broken getRecord method

Also add test covering method.

justusmoroni commented on .travis.yml:
```diff
@@ -25,18 +24,12 @@ env:
 
 matrix:
   fast_finish: true
   exclude:
```
Can't this be removed?

justusmoroni commented on Tests/Unit/Domain/Index/TcaIndexer/TcaTableServiceTest.php:
```diff
@@ -60,10 +60,18 @@ public function doUsePlainQueryIfNoAdditionalWhereClauseIsDefined()
+            ->method('getSystemWhereClause')
+            ->will($this->returnValue('1=1 AND pages.no_search = 0'));
 
+        $whereClause =$this->subject->getWhereClause();
```
Add whitespace after `=`

justusmoroni commented on composer.json:
```diff
@@ -16,12 +16,13 @@
-        "php": ">=5.6.0",
-        "typo3/cms": "~7.6",
+        "php": ">=7.1.0",
+        "typo3/cms": "~8.2",
```
This is correct and works. But shouldn't this be `~8.7`?
Just for good looks?

DanielSiepmann committed:
Merge remote-tracking branch 'origin/develop' into feature/cms-8-support

DanielSiepmann committed:
BUGFIX: Fix broken test

DanielSiepmann committed:
Merge remote-tracking branch 'origin/develop' into feature/cms-8-support

DanielSiepmann committed:
Merge branch 'feature/small-improvements' into feature/cms-8-support

DanielSiepmann committed:
Merge branch 'hotfix/fix-db-tests-on-travis' into feature/cms-8-support

DanielSiepmann committed:
TASK: Remove php 7.0 testing

As we do not support PHP 7.0.

DanielSiepmann committed:
TASK: Fix codacy issues

Break line to not exceed maximum line length.
Use imported namespace to shorten line.

DanielSiepmann committed:
TASK: Cleanup PR issues

DanielSiepmann committed:
Merge remote-tracking branch 'origin/develop' into feature/cms-8-support

Conflicts:
    Configuration/TypoScript/constants.txt

DanielSiepmann committed:
BUGFIX: Do not test with master of TYPO3

This version only supports V8 of TYPO3 CMS.

DanielSiepmann committed:
BUGFIX: Fetch record to update

DanielSiepmann committed:
BUGFIX: Remove pr issue

DanielSiepmann committed:
Merge remote-tracking branch 'origin/develop' into feature/cms-8-support
